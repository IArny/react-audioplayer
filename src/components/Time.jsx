import React, { PropTypes } from 'react';
import style from '../styles/audioElements.css';

const length2two = function(num) {
  if (num === 0) {return '00'; }
  if (num >= 10) { return `${num}`; }
  return `0${num}`;
}

const makeTimeString = function(h, m, s) {
  let ss = length2two(s);
  if (h === 0) {
    return `${m}:${ss}`;
  } else {
    const ms = length2two(m);
    return `${h}:${ms}:${ss}`;
  }
}

const Time = ({ time }, { color }) => {
  let remaining = time;
  const hour = Math.floor(remaining / 3600);
  remaining -= hour * 3600;
  const minute = Math.floor(remaining / 60);
  remaining -= minute * 60;
  const second = Math.floor(remaining);
  const timeString = makeTimeString(hour, minute, second);
  return (
    <p className={style.time}>
      {timeString}
    </p>
  );
};
Time.contextTypes = {
  color: React.PropTypes.string
};

export default Time;
