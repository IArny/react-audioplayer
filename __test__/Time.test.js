import React from 'react';
import Time from '../src/components/Time.jsx';
import renderer from 'react-test-renderer';

test('Time show correct time string when given different currentTime prop', () => {
  const component = renderer.create(
    <Time time={200} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
