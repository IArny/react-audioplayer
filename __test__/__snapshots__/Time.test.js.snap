exports[`test Time show correct time string when given different currentTime prop 1`] = `
<p>
  3:20
</p>
`;
